ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/attachmentpreview/AttachmentPreviewFragmentTest.java

```diff
@@ -0,0 +1,373 @@
+        verify(exitButton).setOnClickListener(any(View.OnClickListener.class));
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenImageUrlNullFilePathNotNullAndShowSendButtonFalseThenSetOnClickListener()
+            throws Exception {
+        setUpForOnActivityCreated();
+        String imageUrl = null;
+        String filePath = "filePath";
+        boolean showSendButton = false;
+        View exitButton = mock(View.class);
+        FileAttachment fileAttachment = mock(FileAttachment.class);
+        TextView attachmentTime = mock(TextView.class);
+        when(FileAttachmentUtil.generateFileAttachment(anyString(), any(File.class))).thenReturn(fileAttachment);
+        when(bundle.getString(anyString())).thenReturn(imageUrl, filePath, "sessionId", "userAgent",
+                "attachment_name", "attachment_download_url");
+        when(bundle.getLong(anyString(), anyLong())).thenReturn(0L, 0L);
+        when(bundle.getBoolean(anyString(), anyBoolean())).thenReturn(true, true, true, true, true, true, true, true);
+        doReturn(showSendButton).when(bundle).getBoolean(KayakoAttachmentPreviewActivity.ARG_SHOW_SEND_BUTTON, false);
+        doReturn(exitButton).when(mRoot).findViewById(R.id.ko__button_exit);
+        doReturn(attachmentTime).when(mRoot).findViewById(R.id.ko__attachment_date);
+
+        underTest.onActivityCreated(bundle);
+        verify(exitButton).setOnClickListener(any(View.OnClickListener.class));
+    }
+
+    @Test
+    public void givenOnActivityCreatedWhenImageUrlAndFilePathIsNullThenIllegalStateException() {
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("INVALID STATE - need at least imageUrl or filePath");
+        KayakoAttachmentPreviewActivity kayakoAttachmentPreviewActivity = mock(KayakoAttachmentPreviewActivity.class);
+        Intent intent = mock(Intent.class);
+        Bundle mockedBundle = mock(Bundle.class);
+        when(underTest.getActivity()).thenReturn(kayakoAttachmentPreviewActivity);
+        doReturn(intent).when(kayakoAttachmentPreviewActivity).getIntent();
+        doReturn(mockedBundle).when(intent).getExtras();
+        underTest.onActivityCreated(bundle);
+    }
+
+    @Test
+    public void givenExtractAuthInfoWhenBundleContainsSessionInfoThenReturnSessionAuth() throws Exception {
+        when(bundle.containsKey(anyString())).thenReturn(true, true);
+        when(bundle.getString(KayakoAttachmentPreviewActivity.ARG_AGENT_SESSION_ID, null)).thenReturn("sessionId");
+        when(bundle.getString(KayakoAttachmentPreviewActivity.ARG_AGENT_USER_AGENT, null)).thenReturn("userAgent");
+        SessionAuth sessionAuth = Whitebox.<SessionAuth>invokeMethod(underTest, "extractAuthInfo", bundle);
+        Assert.assertNotNull(sessionAuth);
+    }
+
+    @Test
+    public void
+    givenConfigureImageForUrlAttachmentShouldSetImageViewAndLoadingViewVisibleAndAttachmentPlaceHolderGone() throws
```

Comment:
⚠️  This snippet of code contains typos. Spelling errors make reading and understanding the source code harder - mild errors cause the reader to stumble for a fraction of a second, many of them can cause considerable delays; really bad errors can render source code completely unreadable.  

**[More details](https://confluence.devfactory.com/x/QAPvEw)**
          
## The possible solution in this case

```JAVA
@Test 
public void 
givenConfigureImageForUrlAttachmentShouldSetImageViewAndLoadingViewVisibleAndAttachmentPlaceholderGone() throws
                                                                                            ^ lower case

```
 

<a id="61a83b90-4faa-11e8-b7bf-a93e6485754a" href="none" codereview-uuid-reserved-tag />